### ISSUE TITLE
Alternate dvclive output folder.

### ISSUE THREAD

**Issue by sebastient** [iterative/vscode-dvc#4485]
The VSCode plugin only looks for plots in the dvclive folder but dvclive does allow custom folders to be provided.  The plugin should look in the actual dvclive output folder for logs and plots.

**Comment by mattseddon** [comment]
Hi @sebastient,

Thanks for raising the issue. 

The extension does not search for plots but instead relies on `dvc plots diff` (docs [here](https://dvc.org/doc/command-reference/plots/diff ) to provide it with the plots for the currently selected experiments. 

Plots have to be declared in one of the `dvc.yaml` files for the DVC project (docs [here](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#plots)). 

If you were already aware of this then sorry for reiterating. Please let me know if I have completely missed the point of the issue.

Thanks,

**Comment by shcheklein** [comment]
@sebastient do you use one of the frameworks' callbacks (Lighting, Keras, etc)? How do you initialize the `Live` instance and pass the custom dir?

I've tried to reproduce it one more time with the example repo that we have:

https://github.com/iterative/example-get-started-experiments/releases/tag/1-notebook-dvclive

(mind the tag - it's a state of the project where one doesn't have any dvc.yaml files in the root, and only dumps metrics with DVCLive into a custom dir)


https://github.com/iterative/vscode-dvc/assets/3659196/d964bd72-3629-4b25-9ac5-396ecafeeeac

Is anything different on your end?

**Comment by sebastient** [comment]
I am using Keras for the model, here is how I initialize live.  I'm not using a context manager, but cleanup the self.live in the class' destructor.

`if dvclive_output is not None:
            from dvclive import Live
            self.live = Live(dvclive_output, report='md', save_dvc_exp=True)
            self.live.log_params(info)
`

Metrics are logged like this, `self.live.log_metric('Learning Rate', lr)`.

It all works fine if I set the dvclive_output above to be `dvclive`, but if I set it up to be `out` then it doesn't show plots in VSCode.

I wasn't aware of having to include plots in dvc.yaml and looking at the documentation it seems I would have to define the plot, whereas right now they just show up and work as expected as long as I set the output to be `dvclive`.

Here's my dvc.yaml file for reference: https://github.com/DeepViewML/playingcards/blob/main/dvc.yaml

**Comment by dberenbaum** [comment]
@sebastient Do you call [`live.end()`](https://dvc.org/doc/dvclive/live/end) anywhere in your code? Could you see if that helps? It should write out the necessary plots configuration. If you aren't using a context manager or calling `live.next_step` or `live.end` anywhere, it's possible that dvclive never configures the plots since it thinks you are still logging the initial step.

**Comment by sebastient** [comment]
I've not been able to reproduce the charting even when moving back to dvclive as the output folder now.  I've created this new example that exercises only the dvc pipeline to make testing easier.  

https://github.com/sebastient/dvcdemo

I'll keep playing around with it to figure out what was happening before, but it seems to be not reproducible now, and according to the documentation I need to manually create plots.  Just to confirm, the [log_metrics](https://dvc.org/doc/dvclive/live/log_metric) with plot=True is not enough for the DVC VSCode plug-in to pick up the charts?

**Comment by dberenbaum** [comment]
@sebastient In https://github.com/sebastient/dvcdemo, the problem is stemming from including `/dvclive` in `.gitignore`. DVC expects this folder to be [git-tracked](https://dvc.org/doc/dvclive/how-it-works#git-integration) to keep track of each run's data.

@daavoo Can we warn or even force add the dvclive folder if it is in .gitignore?

**Comment by sebastient** [comment]
Good catch!  Great, so removing it from `.gitignore` allows graphs to show up automatically in VSCode again.  I can also remove `out` from `.gitignore` and now graphs show up here as well.

I would prefer not to have these saved in Git though, is this a hard requirement?  Which files are needed?  Looking at the generated `out/.gitignore` it seems files tracked by DVC are automatically ignored, but it isn't clear which I need to keep for VSCode integration to work.

**Comment by sebastient** [comment]
So playing around a bit with it and I can ignore `/out` as long as I explicitly un-ignore `/out/dvc.yaml` and `/out/dvc.lock` then all works as expected.

https://github.com/sebastient/dvcdemo/blob/main/.gitignore#L5

I feel like it would be better if `dvc plots` could avoid this restriction but it's pretty minor.  It should though document the requirement as I would expect `/out` or similar folders to typically be in `.gitignore` to avoid polluting git.

Thanks for the help!

**Comment by dberenbaum** [comment]
@sebastient There are a bunch of dvclive issues related to your concerns:

* https://github.com/iterative/dvclive/issues/661 (created as a result of your problems above)
* https://github.com/iterative/dvclive/issues/648
* https://github.com/iterative/dvclive/issues/456

Feel free to chime in there if you have suggestions. Closing this one since it seems it's working for you and the above issues should cover your other concerns.
